public/app/plugins/datasource/grafana-azure-monitor-datasource/models.gen.ts

```diff
@@ -0,0 +1,227 @@
+  resources: [],
+};
+
+export interface AzureResourceGraphQuery {
+  query?: string;
+  resultFormat?: string;
+}
+
+export interface AzureMonitorResource {
+  metricNamespace?: string;
+  region?: string;
+  resourceGroup?: string;
+  resourceName?: string;
+  subscription?: string;
+}
+
+export interface AzureMetricDimension {
+  dimension: string;
+  /**
+   * @deprecated filter is deprecated in favour of filters to support multiselect
+   */
+  filter?: string;
+  filters?: Array<string>;
+  operator: string;
+}
+
+export const defaultAzureMetricDimension: Partial<AzureMetricDimension> = {
+  filters: [],
+};
+
+export type GrafanaTemplateVariableQueryType = ('AppInsightsMetricNameQuery' | 'AppInsightsGroupByQuery' | 'SubscriptionsQuery' | 'ResourceGroupsQuery' | 'ResourceNamesQuery' | 'MetricNamespaceQuery' | 'MetricNamesQuery' | 'WorkspacesQuery' | 'UnknownQuery');
+
+export interface BaseGrafanaTemplateVariableQuery {
+  rawQuery?: string;
+}
+
+export interface UnknownQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'UnknownQuery';
+}
+
+export interface AppInsightsMetricNameQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'AppInsightsMetricNameQuery';
+}
+
+export interface AppInsightsGroupByQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'AppInsightsGroupByQuery';
+  metricName: string;
+}
+
+export interface SubscriptionsQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'SubscriptionsQuery';
+}
+
+export interface ResourceGroupsQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'ResourceGroupsQuery';
+  subscription: string;
+}
+
+export interface ResourceNamesQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'ResourceNamesQuery';
+  metricNamespace: string;
+  resourceGroup: string;
+  subscription: string;
+}
+
+export interface MetricNamespaceQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'MetricNamespaceQuery';
+  metricNamespace?: string;
+  resourceGroup: string;
+  resourceName?: string;
+  subscription: string;
+}
+
+/**
+ * @deprecated Use MetricNamespaceQuery instead
+ */
+export interface MetricDefinitionsQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'MetricDefinitionsQuery';
+  metricNamespace?: string;
+  resourceGroup: string;
+  resourceName?: string;
+  subscription: string;
+}
+
+export interface MetricNamesQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'MetricNamesQuery';
+  metricNamespace: string;
+  resourceGroup: string;
+  resourceName: string;
+  subscription: string;
+}
+
+export interface WorkspacesQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'WorkspacesQuery';
+  subscription: string;
+}
+
+export type GrafanaTemplateVariableQuery = (AppInsightsMetricNameQuery | AppInsightsGroupByQuery | SubscriptionsQuery | ResourceGroupsQuery | ResourceNamesQuery | MetricNamespaceQuery | MetricDefinitionsQuery | MetricNamesQuery | WorkspacesQuery | UnknownQuery);
+
+export interface Azure Monitor {}
```

Comment:
@spinillos could you work out a simpler repro for this buggy output and file an issue (or just a PR?)

Clearly it's an issue with our name sanitizers somewhere, i'm _guessing_ probably [here](https://github.com/grafana/grafana/blob/3b3059c9ce94515baef5c352d07651a404102bfc/pkg/plugins/plugindef/plugindef.go#L43). The right place to put a test case would be [in here](https://github.com/grafana/grafana/blob/3b3059c9ce94515baef5c352d07651a404102bfc/pkg/plugins/pfs/pfs_test.go#L16)

Reply:
Its because cfg.RootName contains spaces. I checked this sanitizer and it doesn't looks like that its causing this issue.

I fixed like this: https://github.com/grafana/thema/pull/104 but idk if its the correct place to do it.

Reply:
Underlying issue here was fixed in #62144 - merging main and rerunning `make gen-cue` should fix it.